commit a4f62448819764f6f27ebcb86115734d0d57ea8d
Author: Sergey V. Udaltsov <svu@gnome.org>
Date:   Sun Sep 2 08:57:52 2012 +0100

    Fixed Shift-F10 behaviour
    
    Preserving shift is essential for many applications
    https://bugs.freedesktop.org/show_bug.cgi?id=45008

diff --git a/types/pc b/types/pc
index 01a5634..daebe74 100644
--- a/types/pc
+++ b/types/pc
@@ -59,17 +59,19 @@ partial default xkb_types "default" {
     };
 
     type "CTRL+ALT" {
-	modifiers = Control+Alt+Shift+LevelThree;
+        modifiers = Control+Alt+Shift+LevelThree;
         map[None] = Level1;
         map[Shift] = Level2;
         map[LevelThree] = Level3;
         map[Shift+LevelThree] = Level4;
-	map[Control+Alt] = Level5;
+        map[Control+Alt] = Level5;
+        preserve[Shift] = Shift;
+        preserve[Shift+LevelThree] = Shift;
         level_name[Level1] = "Base";
         level_name[Level2] = "Shift";
         level_name[Level3] = "Alt Base";
         level_name[Level4] = "Shift Alt";
-	level_name[Level5] = "Ctrl+Alt";
+        level_name[Level5] = "Ctrl+Alt";
     };
 
    // Local eight level
